lib/models/machine.rb

```diff
@@ -231,19 +235,19 @@ def already_submitted?(infrastructure_machines_endpoint)
 
     logger.info "Checking UC6 if #{self.platform_id} was already submitted"
     begin
-      and_query_json = { virtual_name: { eq: self.platform_id } }.to_json
-      response = hyper_client.get(infrastructure_machines_endpoint, {"and": and_query_json})
+      and_query_json = { custom_id: { eq: self.platform_id } }.to_json
+      response = hyper_client.get(infrastructure_machines_endpoint, {"and": and_query_json})# NEED TO CHECK THIS
 
       if response and response.code == 200
-        response_machines_json = response.json["embedded"]["machines"]
+        response_machines_json = JSON.parse(response)["embedded"]["machines"]
```

Comment:
I would use single quotes for the keys here since there is no variable interpolation within them